staging/src/k8s.io/legacy-cloud-providers/azure/azure_loadbalancer.go

```diff
@@ -540,7 +542,7 @@ func (az *Cloud) ensurePublicIPExists(service *v1.Service, pipName string, domai
 	}
 	if len(domainNameLabel) == 0 {
```

Comment:
should assign to nil when the dns annotation is found, `if found && len(domainNameLabel) == 0`